homeassistant/components/ultrasync/__init__.py

```diff
@@ -0,0 +1,170 @@
+"""The Interlogix UltraSync Hub component."""
+
+import asyncio
+
+from ultrasync import AlarmScene
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    SERVICE_AWAY,
+    SERVICE_DISARM,
+    SERVICE_STAY,
+)
+from .coordinator import UltraSyncDataUpdateCoordinator
+
+PLATFORMS = ["sensor"]
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PIN): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: dict) -> bool:
+    """Set up the UltraSync integration."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if hass.config_entries.async_entries(DOMAIN):
+        return True
+
+    if DOMAIN in config:
```

Comment:
Why do you support import from YAML if it's a new integration?

Reply:
I personally like the use of `configuration.yaml` file. :slightly_smiling_face: 

Is it being recommended to not use the YAML any-more? I can't seem to find reference to this path being phased out or deprecated? As per our Discord convo, I was _about_ to drop it as per your advice, but I'm seeing so much reference to this in so many of existing integrations too. I was just curious your reasoning behind it's removal?

Reply:
It is explained on https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md
No more YAML for new integrations.

Other integrations are having to support migration from older setups, but most are gradually phasing it out.